psydac/cad/geometry.py

```diff
@@ -572,8 +570,25 @@ def export_nurbs_to_hdf5(filename, nurbs, periodic=None, comm=None ):
         bounds2 = (float(nurbs.breaks(1)[0]), float(nurbs.breaks(1)[-1]))
         bounds3 = (float(nurbs.breaks(2)[0]), float(nurbs.breaks(2)[-1]))
         domain  = Cube(patch_name, bounds1=bounds1, bounds2=bounds2, bounds3=bounds3)
-
-    mapping = Mapping(mapping_id, dim=nurbs.dim)
+    
+    degrees = nurbs.degree
+    points=nurbs.points[...,:nurbs.dim]
+    knots=[]
+    for d in range( nurbs.dim ):
+        knots.append(nurbs.knots[d])
+    if periodic is None:
+        periodic=[False for d in range( nurbs.dim )]
+    if rational:
+        weights = nurbs.weights
+    spaces=[SplineSpace(knots=k,degree=p) for k,p in zip(knots, degrees)]
+    ncells = [len(space.breaks)-1 for space in spaces]
+    domain_decomposition = DomainDecomposition(ncells,periodic,comm)
+    space=TensorFemSpace(domain_decomposition,*spaces)
+    if rational:
+        mapping = NurbsMapping.from_control_points_weights(space, points, weights)
+    else:
+        mapping=SplineMapping.from_control_points(space, points)
+    print("domain=mapping(domain)")    
```

Comment:
Please apply PEP8 guidelines on style:
```suggestion
    degrees = nurbs.degree
    points = nurbs.points[..., :nurbs.dim]
    knots = []
    for d in range(nurbs.dim):
        knots.append(nurbs.knots[d])
    if periodic is None:
        periodic = [False] * nurbs.dim
    if rational:
        weights = nurbs.weights
    spaces = [SplineSpace(knots=k, degree=p) for k, p in zip(knots, degrees)]
    ncells = [len(space.breaks) - 1 for space in spaces]
    domain_decomposition = DomainDecomposition(ncells, periodic, comm)
    space = TensorFemSpace(domain_decomposition, *spaces)
    if rational:
        mapping = NurbsMapping.from_control_points_weights(space, points, weights)
    else:
        mapping = SplineMapping.from_control_points(space, points)
    print("domain=mapping(domain)")
```